File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) == 1:
            return 1
        mySet=set()
        highest=0
        l,r=0,0

        for r in range(len(s)):
           
            while s[r] in mySet:
                mySet.remove(s[l])
                l+= 1

    
            mySet.add(s[r])
            highest=max(highest,len(mySet))
        return highest


            
      


        

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cs


public class Solution {
    public int LengthOfLongestSubstring(string s) {

        int l=0;
        HashSet<int> myHash= new HashSet<int>();
        int longest=0;
        
        for(int r=0;r<s.Length;r++){

            while(myHash.Contains(s[r])){
               
                myHash.Remove(s[l]);
                l ++;

            }
            myHash.Add(s[r]);
            longest=Math.Max(longest,myHash.Count);


        }
        return longest;
        
    }
}